### ISSUE TITLE
[asyncftpclient] allow interaction with the data connection

### ISSUE THREAD

**Issue by samdmarshall** [nim-lang/Nim#11023]
right now it seems impossible to interact directly with any data-connection using asyncftpclient as the methods `pasv` and `getLines` are not exported publicly. this gets wrapped internally for a couple commands but outside of that it seems like you would have to reimplement the whole module or copy it into your code to customize and access these components.

**Comment by Araq** [comment]
Is it as easy as exporting these procs?

**Comment by samdmarshall** [comment]
that is my impression; i copied the file into my project dir and added the export and things seemed to “just work”; you have to invoke the `pasv` call prior to calling the `send`, await that future and then await the `getLines` call. wanted to run this past y’all before throwing up a PR for it.

**Comment by dom96** [comment]
Why do you want to interact with the data connection directly?

**Comment by samdmarshall** [comment]
some of the commands i was trying to use resulted in the server opening a data connection to transmit the response. maybe instead of exporting those two procs, a better approach might be to alter the `send` to pass back data from the data-connection if that gets triggered?

**Comment by dom96** [comment]
yes, that sounds better although I'm still not certain of the details so can't advise 100%
